src/main/kotlin/com/gamingbacklog/api/gamingbacklogapi/clients/IGDBClient.kt

```diff
@@ -54,23 +53,18 @@ class IGDBClient {
     gameID: String
   ): Array<IGDBGame> {
     val body = "fields name, platforms.name, genres.name, release_dates.human, summary, franchises.name, involved_companies.company.name, artworks.url; where id =$gameID;"
-    val request: HttpRequest = HttpRequest.newBuilder()
-      .uri(URI.create("${baseUri}games/"))
-      .POST(HttpRequest.BodyPublishers.ofString(body))
-      .header("Client-ID", clientId)
-      .header("Authorization", "Bearer $accessToken")
-      .build()
-    try {
-      val response = HttpClient.newHttpClient().send(request, HttpResponse.BodyHandlers.ofString())
-      println("Request: ${request.uri()} Body: ${request.bodyPublisher()}")
-      println("Response: ${response.body()}")
-      // todo: need better error handling here for 404
-      return Gson().fromJson(response.body(), Array<IGDBGame>::class.java)
-    } catch (ex: Exception) {
-      ex.printStackTrace()
+    val headers = HashMap<String, String>()
+    headers["Client-ID"] = clientId
+    headers["Authorization"] = "Bearer $accessToken"
+    val response = externalAPICall.postExternalCall("${baseUri}games/", body, headers)
```

Comment:
What is this call actually doing? Do we have to authenticate in the same call in order to fetch the games?